db/migrate/20150930183738_migrate_content_hosts.rb

```diff
@@ -1,14 +1,19 @@
 class MigrateContentHosts < ActiveRecord::Migration
-  class Katello::HostCollectionAssociation < ActiveRecord::Base
+  class Katello::SystemRepository < ActiveRecord::Base
+    self.table_name = "katello_system_repositories"
+    belongs_to :system, :inverse_of => :system_repositories, :class_name => 'Katello::System'
+    belongs_to :repository, :inverse_of => :system_repositories, :class_name => 'Katello::Repository'
+  end
+
+  class Katello::SystemHostCollections < ActiveRecord::Base
     self.table_name = "katello_system_host_collections"
   end
 
-  class Katello::ErrataAssociation < ActiveRecord::Base
+  class Katello::SystemErratum < ActiveRecord::Base
     self.table_name = "katello_system_errata"
-  end
 
-  class Katello::RepositoryAssociation < ActiveRecord::Base
-    self.table_name = "katello_system_repositories"
+    belongs_to :system, :inverse_of => :system_errata, :class_name => 'Katello::System'
+    belongs_to :erratum, :inverse_of => :system_errata, :class_name => 'Katello::Erratum'
   end
 
   class Katello::System < ActiveRecord::Base
```

Comment:
This is problematic in that system has a type column so even if we try to redefine another class (e.g. MyMigrationSystem), calls to the database (e.g. MyMigrationSystem.first) are still going to return Katello::System objects. The only thing I can think of is call `becomes` on everything that comes out of the database (e.g. MyMigrationSystem.first.becomes(MyMigrationSystem)). Thoughts?